sgx-sys/src/lib.rs

```diff
@@ -114,8 +82,51 @@ mod test {
             0,
         );
 
-        let secs = Secs::new(8192, 8192, 4096, &contents);
-
+        let secs = secs::Secs::new(0, secs::Secs::SIZE_MAX, 4096, &contents);
         Builder::new(secs).unwrap();
     }
+
+    #[cfg_attr(not(has_sgx), ignore)]
+    #[test]
+    fn builder_add() {
+        let contents = sig::Contents::new(
+            misc::MiscSelect::default(),
+            attr::Attributes::default(),
+            [0u8; 32],
+            0,
+            0,
+        );
+        let secs = secs::Secs::new(0, secs::Secs::SIZE_MAX, 4096, &contents);
+        let mut builder = Builder::new(secs).unwrap();
+
+        let flags = [Flags::empty(), Flags::MEASURE];
+        let perms = [
+            page::Flags::empty(),
+            page::Flags::R,
+            page::Flags::R | page::Flags::W,
+            page::Flags::R | page::Flags::X,
+            page::Flags::R | page::Flags::W | page::Flags::X,
+        ];
+
+        let mut off = 0;
+        for f in &flags {
+            #[repr(C, align(4096))]
+            struct Page([u8; 4096]);
+            let page = Page([0u8; 4096]);
+
+            // TCS pages MUST NOT specify permissions.
+            let si = page::SecInfo::new(page::Flags::empty(), page::Class::Tcs);
+            eprintln!("{:?} {:?} {:?}", off, *f, si);
+            assert!(unsafe { builder.add(off, &page.0, *f, si).is_ok() });
```

Comment:
Is this asserting that the builder was able to add a TCS page with the particular flag (`empty` for TCS) and secinfo combination at an offset that advances by one page each time?

Reply:
No. `TCS` flags have no permissions. Therefore, the set of flags is empty.